packages/react-devtools-shared/src/backend/renderer.js

```diff
@@ -3747,7 +3795,57 @@ export function attach(
   }
 
   // React will switch between these implementations depending on whether
-  // we have any manually suspended Fibers or not.
+  // we have any manually suspended/errored-out Fibers or not.
+  function shouldErrorFiberAlwaysNull() {
+    return null;
+  }
+
+  const forceErrorForFiberIDs = new Map();
+  function shouldErrorFiberAccordingToMap(fiber) {
+    if (typeof setErrorHandler !== 'function') {
+      throw new Error(
+        'Expected overrideError() to not get called for earlier React versions.',
+      );
+    }
+
+    const id = getOrGenerateFiberID(fiber);
+    let status = null;
+    if (forceErrorForFiberIDs.has(id)) {
+      status = forceErrorForFiberIDs.get(id);
+      if (status === false) {
+        clearErrorsForFiberID(id);
+      }
+
+      if (forceErrorForFiberIDs.size === 0) {
+        // Last override is gone. Switch React back to fast path.
+        setErrorHandler(shouldErrorFiberAlwaysNull);
+      }
+    }
+    return status;
+  }
+
+  function overrideError(id, forceError) {
+    if (
+      typeof setErrorHandler !== 'function' ||
+      typeof scheduleUpdate !== 'function'
+    ) {
+      throw new Error(
+        'Expected overrideError() to not get called for earlier React versions.',
+      );
+    }
+
+    forceErrorForFiberIDs.set(id, forceError);
+
+    if (forceErrorForFiberIDs.size === 1) {
+      // First override is added. Switch React to slower path.
+      setErrorHandler(shouldErrorFiberAccordingToMap);
+    }
+
+    const fiber = idToArbitraryFiberMap.get(id);
+    if (fiber != null) {
+      scheduleUpdate(fiber);
+    }
+  }
```

Comment:
Doesn't look like we ever clean up entries here.

If `forceError` is `false` shouldn't we `forceErrorForFiberIDs.delete()` and then maybe reset the error handler (if the map is now empty)?

Reply:
oh, cleaning it up happens in `shouldErrorFiberAccordingToMap` since if we clean it up here, toggling the error won't work

Reply:
I needed to have the clean up in `shouldErrorFiberAccordingToMap` unfortunately because we are relying on the value `false` to know when to toggle off. If we delete it here right away, when `shouldErrorFiberAccordingToMap` is called, it will not know.

Is there a way to "schedule" for the ID to be cleaned up after the toggling is done?

Reply:
> If we delete it here right away, when shouldErrorFiberAccordingToMap is called, it will not know.

I don't understand. Why wouldn't it know?

An ID not being inside of `shouldErrorFiberAccordingToMap` is the same as it being in there with a value of `false` as far as React's behavior, isn't it?

Reply:
Put another way, `shouldErrorFiberAccordingToMap` could just be a set of IDs. Anything in the Set is "true"

Reply:
Ah yeah, this is from [our previous discussion](https://github.com/bvaughn/react/commit/23236e68c359b8b29db4543776810d2bd23d8ed2), we used a set before and toggling error-state off didn't work then. So we had to use a map to have 3 different values: true/false/null (`true`: error, `false`: toggle off, `null`: do nothing)

Reply:
Oh! That's right. Sorry. I forgot how landed on that before. My apologies!

You're right 😁  For now, it's okay to clear on read like you're doing. Maybe we'll revisit that in the future if we think of a different way!